segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentContainerRegistryTests.java

```diff
@@ -207,28 +241,31 @@ public int getId() {
 
         @Override
         public void close() {
-            if (!this.closed) {
+            if (!this.closed.getAndSet(true)) {
                 FutureHelpers.await(ServiceHelpers.stopAsync(this, ForkJoinPool.commonPool()));
-                this.closed = true;
             }
         }
 
         @Override
         protected void doStart() {
-            if (this.startException != null) {
-                notifyFailed(this.startException);
-            } else {
-                notifyStarted();
-            }
+            ForkJoinPool.commonPool().execute(() -> {
```

Comment:
Guava's Service doc instructs the `doStop`/`doStart` methods to execute their code async and return immediately, and only call `notifyStarted`/`notifyStopped`/`notifyFailed` when they are done. If not, then the whole start/stop would be synchronous. I just changed the unit test "Container" to execute stuff asynchronously - otherwise the condition that we are testing would never be triggered.